nbgrader_to_canvas/upload_grades.py

```diff
@@ -0,0 +1,379 @@
+        return assignment
+
+    def _get_max_score(self):
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            nb_assignment = gb.find_assignment(self._form_nb_assign_name)
+            return nb_assignment.max_score
+
+    def _get_num_students(self):
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            return len(gb.students)
+
+    def _create_assignment(self, max_score):
+        return self._course.create_assignment({'name':self._form_nb_assign_name, 'published':'true', 'assignment_group_id':self._group, 'points_possible':max_score})
+        
+    def _delete_comments(self):
+        '''clears old comments from submission to prevent cluttering of the comment section in case of re-submission'''
+        update_status(self._form_nb_assign_name, self._course_id, completion=5, status='Removing Old Feedback')
+        counter = 0
+        auth_header = {'Authorization': 'Bearer ' + self._flask_session['api_key']}
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            for student in gb.students:
+                try:
+                    submission = self.assignment_to_upload.get_submission(self.canvas_students[student.id],include=['submission_comments'])
+                    
+                    for comment in submission.submission_comments:
+                        if comment['comment'] == 'Transferred from datahub.':
+                            response = requests.delete(
+                                "{}api/v1/courses/{}/assignments/{}/submissions/{}/comments/{}".format(
+                                    settings.API_URL, self._course_id, self.assignment_to_upload.id, student.id, comment['id']
+                                ),
+                                headers = auth_header
+                                )
+                except Exception as ex:
+                    raise ex
+                    app.logger.error('Error deleting comments: {}'.format(ex))
+                if counter < self._num_students:
+                    counter += 1
+                    update_status(self._form_nb_assign_name, self._course_id, completion=5+15*counter/self._num_students)
+            
+
+    def _get_feedbacks(self):
+        update_status(self._form_nb_assign_name, self._course_id, completion=20, status='Fetching Feedback')
+        counter = 0
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            for student in gb.students:
+                if not os.path.isdir('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name):
+                    continue
+                for file in os.listdir('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name):
```

Comment:
The same path is used multiple times - assign to a variable? It would help with readability, too

Reply:
n/m, it's not - there's a pull request: https://github.com/ucfopen/canvasapi/pull/510

Reply:
do this